posts/saving-and-analyzing-trending-topics-on-twitter-using-aws-athena-lambda-and-cdk.md

```diff
@@ -0,0 +1,198 @@
+
+```python
+from aws_cdk import (
+    aws_s3 as s3,
+    aws_events as events,
+    aws_lambda as lambda_,
+    aws_events_targets as targets,
+    core,
+)
+
+
+class TwitterTrendsStack(core.Stack):
+
+    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        bucket = s3.Bucket(self, 'tlebrun-playground')
+
+        handler = lambda_.Function(self, 'TrendsHandler',
+                                   runtime=lambda_.Runtime.PYTHON_3_8,
+                                   code=lambda_.Code.asset('resources'),
+                                   handler='trends.main',
+                                   environment=dict(BUCKET=bucket.bucket_name,
+                                                    CONSUMER_KEY='',
+                                                    CONSUMER_SECRET='',
+                                                    ACCESS_TOKEN_KEY='',
+                                                    ACCESS_TOKEN_SECRET='',
+                                                    WOEID='2459115')
+                                   )
+
+        bucket.grant_read_write(handler)
+
+        events.Rule(self, 'RuleHandler',
+                    rule_name='Twitter_Trends_15_Minutes_Rule',
+                    schedule=events.Schedule.rate(duration=core.Duration.minutes(15)),
+                    targets=[targets.LambdaFunction(handler)])
+```
+
+The project is available at this [GitHub repository](https://github.com/Falydoor/twitter-trends) in case you want to clone it.
+
+# Data Visualization with AWS Athena
+
+## Database and table creation
+
+Once you have the lambda running for few days, you will be able to view the data in few minutes using [AWS Athena](https://aws.amazon.com/athena/). Athena lets you query your data stored on S3 without the burden of setting up a whole database and having batch processes running. You can create a new database using the AWS Glue service using the console and then create the table using the Athena service.
+
+```sql
+-- Create table
+CREATE EXTERNAL TABLE
+IF NOT EXISTS twitter.trend
+(
+  `name` string,
+  `url` string,
+  `promoted_content` string,
+  `query` string,
+  `tweet_volume` bigint
+)
+ROW FORMAT SERDE 'org.openx.data.jsonserde.JsonSerDe'
+WITH SERDEPROPERTIES (
+  'serialization.format' = '1'
+) LOCATION 's3://S3_BUCKET/'
+TBLPROPERTIES
+('has_encrypted_data'='false');
+```
+
+The query above will create the table, the name of the fields are the same as the one from the JSON stored on S3. Don't forget to replace `S3_BUCKET` with the actual bucket containing the files.
```

Comment:
The query above will create the table; the name of the fields are the same as the one from the JSON stored on S3.